x-pack/platform/plugins/shared/onechat/server/services/agents/conversational/handler.ts

```diff
@@ -30,54 +25,29 @@ export const createHandler = ({
     { agentParams: { nextInput, conversation = [] }, runId },
     { request, modelProvider, toolProvider, events, runner }
   ) => {
-    const model = await modelProvider.getDefaultModel();
-    const tools = await providerToLangchainTools({ request, toolProvider, runner, logger });
-    const initialMessages = conversationLangchainMessages({
-      nextInput,
-      previousRounds: conversation,
-    });
-    const agentGraph = await createAgentGraph({ logger, chatModel: model.chatModel, tools });
-
-    const eventStream = agentGraph.streamEvents(
-      { initialMessages },
+    const completedRound = await runChatAgent(
       {
-        version: 'v2',
-        runName: defaultAgentGraphName,
-        metadata: {
-          graphName: defaultAgentGraphName,
-          runId,
+        nextInput,
+        conversation,
+        agentGraphName: defaultAgentGraphName,
+        runId,
+        onEvent: (event) => {
+          events.emit(event);
         },
-        recursionLimit: 10,
-        callbacks: [],
+        tools: toolProvider,
```

Comment:
will there be a way to pass in a particlar list of tools in a toolProvider? ie filter tools within the toolProvider? 

Would it be possible to provide tools from multiple providers?

Reply:
So atm you can either pass a provider, or a list of executable tools (which is what is returned from a provider):

https://github.com/elastic/kibana/blob/66b7d9a2157cfcc1e0a06028f08bfc4a47a2c03e/x-pack/platform/plugins/shared/onechat/server/services/agents/conversational/run_chat_agent.ts#L49-L53

In theory it allows to compose however you want (by picking tools, or composing providers, or whatever). The DX may not be ideal though, we probably want some utils (e.g. to allow picking tools from a provider based on ids, things like that)